Os 782 remove the contracts ethers and do rollup in mini sdk

mathewmeconry opened the pull request.

mathewmeconry committed:
refactor(OS-782): removes contracts-ethers package

Removes a now unecessary contracts-ethers package. js-client creates its
own typechain now

BREAKING CHANGE: Removal of contracts-ethers

mathewmeconry committed:
refactor(OS-782): js-client generates typescript

js-client doesn't depend on contracts-ethers anymore and creates the
typechain typings during its own build

josemarinas reviewed (approved):
LGTM, tests are failing because of linting but i guess thats the matter of another PR

mathewmeconry committed:
test(OS-782): ganache replacment, optimizations, use native jest

replaces depricated ganache setup with hardhat for testing, removes
contract interactions where they aren't necessary, upgrades jest by not
using tsdx for testing

mathewmeconry committed:
ci(OS-782): fixes js-client publish and tests workflows

corrects reusable worfklow in js-client publishing workflow, adapts
js-client-tests to use the new typechain integration properly

mathewmeconry committed:
build(OS-782): removes contracts-ethers linting command

mathewmeconry committed:
chore(OS-782): applies prettier formatting

mathewmeconry commented:
Unfortunately I cannot fix the linting errors without replacing tsdx. That this pull doesn't get any bigger I'll do that in a different branch.

The problem is that tsdx is unmaintained and uses babel-eslint which breaks with the latest version of eslint. And we need the latest version of eslint so that we can upgrade typescript to v5 that the contracts build while tsdx is stuck with typescript v4 ....
